libclamav/cvd.c

```diff
@@ -627,6 +630,14 @@ cl_error_t cli_cvdload(FILE *fs, struct cl_engine *engine, unsigned int *signo,
 
     cli_dbgmsg("in cli_cvdload()\n");
 
+    // FIPS verification MUST preclude other verification if in FIPS mode.
+    if (cli_get_fips_mode()) {
+        ret = cli_sigver_external(filename);
```

Comment:
This call should not be guarded by cli_fips_mode, external signature file should be preferred if present.

Reply:
I concur. If we're going to add a better way to validate signing, it should be preferred.  We should only fallback to the legacy signature verification if A. the new-style signature is not present and B. not FIPS-mode.

Reply:
Rather than adding comments all over the code to this extent. Every place that guards the validation/fetching of the external signature file with cli_fips_mode should be removed so that the new format is the default and only trusted signature in versions that support it.